packages/preact/src/components/SearchInput.tsx

```diff
@@ -0,0 +1,29 @@
+import { isHTMLInput } from "@preact/dom/utils"
+import { ComponentChildren } from "preact"
+
+import { renderHeadless } from "../dom/renderHeadless"
+
+type Props = {
+  children: ComponentChildren
+  onSearchInput: (target: HTMLInputElement) => void
+}
+
+export function SearchInput({ children, onSearchInput }: Props) {
+  return renderHeadless({
+    children,
+    updateElement: vnode => {
+      if (!isHTMLInput(vnode) || vnode.props.type !== "search") {
+        return vnode
+      }
+      vnode.props = {
+        ...vnode.props,
+        onInput: event => {
+          if (event.target instanceof HTMLInputElement) {
+            onSearchInput(event.target)
+          }
+        }
+      }
+      return vnode
```

Comment:
Isn't objects are updated by reference. This should automatically reflect in the caller method. Do we need this object returned?